mscore/palette.cpp

```diff
@@ -730,12 +782,69 @@ void Palette::leaveEvent(QEvent*)
       {
       if (currentIdx != -1) {
             QRect r = idxRect(currentIdx);
-            currentIdx = -1;
+            PaletteBox* pb = mscore->getPaletteBox();
+            if (!filterActive || !pb->getKeyboardNavigation())
+                  currentIdx = -1;
             update(r);
             }
       }
 
 //---------------------------------------------------------
+//   nextPaletteElement
+//---------------------------------------------------------
+void Palette::nextPaletteElement()
+      {
+      if (!filterActive)
```

Comment:
Good point.  Checking to see if the number of filtered cells is equal to the total number of cells is not a good enough way to tell if the filter is active or not.